web/tests/test_self_report_attendance.py

```
@@ -0,0 +1,198 @@
+        self.future_session = Session.objects.create(
+            course=self.course,
+            title="Future Session",
+            description="This session will happen tomorrow",
+            start_time=now + timedelta(days=1),
+            end_time=now + timedelta(days=1, hours=2),
+        )
+
+        # Set up client
+        self.client = Client()
+
+    def test_login_required(self):
```

Ruff (0.8.2)

Missing return type annotation for public function `test_login_required`

Add return type annotation: `None`

(ANN201)